cmd/serve/s3/auth.go

```diff
@@ -0,0 +1,25 @@
+package s3
+
+import (
+	"net/http"
+
+	"github.com/rclone/rclone/cmd/serve/s3/signature"
+)
+
+func (p *Server) authMiddleware(handler http.Handler) http.Handler {
```

Comment:
Would it be possible to make the auth proxy work with this server?

https://github.com/rclone/rclone/blob/master/cmd/serve/proxy/proxy.go

Reply:
Changed. But I'm not sure if this will work with the s3 test.